python/mdvtools/auth/auth0_provider.py

```
@@ -180,18 +190,38 @@ def handle_callback(self) -> Optional[str]:
+            # Decode the token's header to inspect its algorithm and other details
+            try:
+                header = jwt.get_unverified_header(access_token)
+                #print("Token Header:", header)  # Check the header of the token
+
+                # Ensure the algorithm is RS256 (not JWE)
+                if header.get('alg') != 'RS256':
+                    logging.error(f"Expected RS256 algorithm, but found {header.get('alg')}")
+                    raise ValueError("The token is not of type RS256.")
+            except Exception as e:
+                logging.error(f"Error decoding the token header: {e}")
+                raise ValueError("Invalid token format.")
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)